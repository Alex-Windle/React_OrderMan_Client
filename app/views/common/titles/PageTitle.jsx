/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';


/**
 * PageTitle represents a title at the top of each authenticated page that
 * tells the user where they are.
 *
 * @description Renders a heading element that takes:
 *   title
 *
 * @extends Component
 */
export default class PageTitle extends Component {
  render() {
    const { title } = this.props;
    return <h1>{title}</h1>;
  }
}


PageTitle.propTypes = {
  title: PropTypes.string.isRequired,
};


PageTitle.defaultProps = {};
