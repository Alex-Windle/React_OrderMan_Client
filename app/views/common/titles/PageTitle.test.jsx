/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React from 'react';
import renderer from 'react-test-renderer';


// Component imports
import PageTitle from './PageTitle';


describe('PageTitle component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <PageTitle title="Sample Title" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
