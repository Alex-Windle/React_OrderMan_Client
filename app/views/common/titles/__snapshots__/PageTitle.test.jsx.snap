// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle component matches snapshot 1`] = `
<h1>
  Sample Title
</h1>
`;
